src/_pytest/compat.py

```diff
@@ -357,6 +358,19 @@ def overload(f):  # noqa: F811
         return f
 
 
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 8):
+        from typing import final as final
+    else:
+        from typing_extensions import final as final
+elif sys.version_info >= (3, 8):
+    from typing import final as final
```

Comment:
@bluetech  What is the reason for using `from typing import final as final` instead of just `from typing import final`?

Reply:
This is a common pattern if you import something deliberately to re-export it (rather than being an unused import for inside this module only) - see e.g. https://mypy.readthedocs.io/en/stable/config_file.html#confval-implicit_reexport